proxy test fails: TestBarrierSingleNode

```
=== RUN   TestBarrierSingleNode
--- FAIL: TestBarrierSingleNode (0.70s)
	v3_barrier_test.go:74: barrier timed out
=== RUN   TestBarrierMultiNode
--- FAIL: TestBarrierMultiNode (1.19s)
	v3_barrier_test.go:74: barrier timed out
=== RUN   TestDoubleBarrier
```

More logging info https://jenkins-etcd.prod.coreos.systems/job/etcd-proxy/181/consoleText

--- a/proxy/grpcproxy/watch_broadcast.go
+++ b/proxy/grpcproxy/watch_broadcast.go
@@ -89,8 +89,8 @@ func (wb *watchBroadcast) bcast(wr clientv3.WatchResponse) {
 	for r := range wb.receivers {
 		r.send(wr)
 	}
-	if wb.size() > 0 {
-		eventsCoalescing.Add(float64(wb.size() - 1))
+	if len(wb.receivers) > 0 {
+		eventsCoalescing.Add(float64(len(wb.receivers) - 1))
 	}
 }
 
@@ -135,7 +135,7 @@ func (wb *watchBroadcast) delete(w *watcher) {
 		panic("deleting missing watcher from broadcast")
 	}
 	delete(wb.receivers, w)
-	if !wb.empty() {
+	if len(wb.receivers) > 0 {
 		// do not dec the only left watcher for coalescing.
 		watchersCoalescing.Dec()
 	}

